vocabularies/Common.xml

```diff
@@ -1425,6 +1425,14 @@ If the request succeeds the response will contain a JSON body of `Content-Type:
         <Annotation Term="Core.Description" String="Base URL for WebSocket connections" />
       </Term>
 
+      <Term Name="ValidationFunction" Type="Common.QualifiedName" Nullable="false" AppliesTo="Parameter Action Function">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Function that validates whether the parameter value, or parameter values (in case of Action and Function) is correct or not" />
+        <Annotation Term="Core.LongDescription">
+          <String>This annotation must not target a binding parameter. When it targets a non-binding parameter, the validation function has one parameter to which the to-be-validated parameter value must be passed. When this annotation targets an action or function, the validation function has one parameter per non-binding action/function parameter to which the to-be-validated value must be passed. The validation function returns true or false.</String>
```

Comment:
```suggestion
          <String>When this annotation targets a non-binding parameter, the validation function must have an unbound overload with one parameter to which the to-be-validated parameter value must be passed. When this annotation targets an action or function, the validation function must have an overload with the same parameter types (including binding parameter, if applicable) as the annotation target. The validation function returns true or false.</String>
```

Reply:
You lost me as soon as you introduced the word overload here.
Why would overload matter in this case ? We're talking about a function such as

```myValidationFunction(parameter:any): 200OK | Error with message which probably need to be defined / aligned`

Reply:
An action or function can have several overloads with different parameters. And you could annotate individual overloads with different validation functions, or with different overloads of the same validation function.

> `myValidationFunction(parameter:any): 200OK | Error with message which probably need to be defined / aligned`

This is a validation function for _one_ parameter. But as @uhlmannm pointed out, the result of the validation can also depend on other parameters, including the binding parameter.

Reply:
Ok in which case if you're targeting a bound action/ function you also need the validation function to exist in relation to the existing binding parameter, make sense

Reply:
A validation function needs a binding parameter, if it shall validate more than one parameter of a bound action or function. For validating only _one_ parameter, an unbound validation function is needed.

Reply:
I see only a limited use in a validation without the binding parameter. A validation from business perspective would often require to know the context.
Also for the UI design I would not see "bound validation functions" as a big restriction. Normally, the user selects the object / binding parameter, then presses the action button and finally fills all the other parameters in a popup. So the binding parameter is available when calling a validation function.

Reply:
> I see only a limited use in a validation without the binding parameter.

I have updated my suggestion accordingly.